### ISSUE TITLE
RBAC Support

### ISSUE THREAD

**Issue by camille-rodriguez** [canonical/mlflow-operator#45]
Feature request to have MLFlow Operator support RBAC. Currently, with the deployment of Charmed MLFlow and MariaDB, all users can see other users data. It would be ideal to be able to segregate data access.

**Comment by gaure** [comment]
Thanks Camille.
We have a similar situation.
We split our data scientist in 2 different teams. 
The training artifacts (metrics, checkpoints, numpy array, etc) generated by team 1 during a RUN execution, and stored in MLFlow under that RUN, cannot be access by team 2, and vise verse.  
It seems that an MLFLow authorization plug-in does not exist.
Are there any plans to implement it?

**Comment by gaure** [comment]
Thanks a lot for the feedback @DomFleischmann

**Comment by ca-scribner** [comment]
I haven't looked into this in a while, but in the past I believe user isolation for metadata within MLFlow was not possible.  iirc, at least in the open source version of mlflow, there were no provisions for this.  I've seen this solved by deploying multiple instance of mlflow, one for each team/isolation group.  I think a mixture of that plus some RBAC control or istio (authorization policies?) could effectively achieve this

**Comment by andreeamun** [comment]
@ca-scribner , @i-chvets - is this something you looked at after?

@misohu , will the new version of MLFlow cover this? cc: @marowid  @Barteus

**Comment by camille-rodriguez** [comment]
@ca-scribner I do not think that this is true. Upstream mlflow allows for RBAC segregation, this is the current setup in one external environment.

**Comment by i-chvets** [comment]
@andreeamun I don't think we have RBAC in our spec for new MLflow. At least not explicitly.

**Comment by andreeamun** [comment]
@camille-rodriguez , I just checked, but it is not supported in the upstream yet either: https://github.com/mlflow/mlflow/issues/7347
